app/uk/gov/hmrc/cdsreimbursementclaimfrontend/views/claims/enter_no_legacy_mrn.scala.html

```diff
@@ -0,0 +1,71 @@
+@*
+* Copyright 2021 HM Revenue & Customs
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*@
+@import play.api.data.Form
+@import play.api.i18n.Messages
+@import play.twirl.api.Html
+@import uk.gov.hmrc.cdsreimbursementclaimfrontend.config.ViewConfig
+@import uk.gov.hmrc.cdsreimbursementclaimfrontend.controllers.claims.routes
+@import uk.gov.hmrc.cdsreimbursementclaimfrontend.controllers.actions.RequestWithSessionData
+@import uk.gov.hmrc.cdsreimbursementclaimfrontend.models.MovementReferenceNumber
+@import uk.gov.hmrc.cdsreimbursementclaimfrontend.views.utils.LanguageHelper._
+
+@this(
+        layout: uk.gov.hmrc.cdsreimbursementclaimfrontend.views.html.Layout,
+        pageHeading: uk.gov.hmrc.cdsreimbursementclaimfrontend.views.html.components.page_heading,
+        paragraph: uk.gov.hmrc.cdsreimbursementclaimfrontend.views.html.components.paragraph_block,
+        inputText: uk.gov.hmrc.cdsreimbursementclaimfrontend.views.html.components.input_text,
+        submitButton: uk.gov.hmrc.cdsreimbursementclaimfrontend.views.html.components.button,
+        errorSummary: uk.gov.hmrc.cdsreimbursementclaimfrontend.views.html.components.error_summary,
+        backLinkComponent: uk.gov.hmrc.cdsreimbursementclaimfrontend.views.html.components.back_link,
+        formWithCSRF: uk.gov.hmrc.govukfrontend.views.html.helpers.formWithCSRF
+)
+
+@(form: Form[MovementReferenceNumber], isAmend: Boolean, subKey:Option[String])(implicit request: RequestWithSessionData[_], messages: Messages, viewConfig: ViewConfig)
+
+@key = @{ "enter-no-legacy-mrn" }
+
+@title = @{ messages(lang(key,subKey,"title")) }
+
+@postAction = @{ if(isAmend) routes.EnterMovementReferenceNumberController.changeMrnSubmit else routes.EnterMovementReferenceNumberController.enterMrnSubmit }
+
+@layout(pageTitle = Some(s"$title"), signedInUserDetails = request.signedInUserDetails) {
+
+    @if(isAmend) { @backLinkComponent(routes.CheckYourAnswersAndSubmitController.checkAllAnswers) }
+
```

Comment:
we shouldn't have this type of logic anymore in the service - back-links simulate the back-browser button.

Reply:
Done